internal/pkg/api/handleCheckin.go

```diff
@@ -471,11 +471,15 @@ func (ct *CheckinT) processUpgradeDetails(ctx context.Context, agent *model.Agen
 			vSpan.End()
 			break // no validation
 		}
-		_, err := details.Metadata.AsUpgradeMetadataDownloading()
+		upgradeDetails, err := details.Metadata.AsUpgradeMetadataDownloading()
 		if err != nil {
 			vSpan.End()
 			return fmt.Errorf("%w %s: %w", ErrInvalidUpgradeMetadata, UpgradeDetailsStateUPGDOWNLOADING, err)
 		}
+		if err := details.Metadata.FromUpgradeMetadataDownloading(upgradeDetails); err != nil {
+			vSpan.End()
+			return fmt.Errorf("%w %s: unable to repack metadata: %w", ErrInvalidUpgradeMetadata, UpgradeDetailsStateUPGDOWNLOADING, err)
```

Comment:
Sorry for the late review, but the root cause one layer up from this is we consider parsing upgrade details to be a fatal error and it isn't. Upgrade details are nice to have. 

- https://github.com/elastic/fleet-server/issues/5199